File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/13_가장많이사용된자릿수/main.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func main() {
	file, _ := os.Open("input.txt")
	defer file.Close()
	old := os.Stdin
	defer func() { os.Stdin = old }()

	os.Stdin = file
	var N int
	var arr [10]int
	fmt.Scanf("%d", &N)

	for _, v := range strconv.Itoa(N) {
		val, _ := strconv.Atoi(string(v))
		arr[val]++
	}
	var max int
	var val int
	for i, v := range arr {
		if v > max {
			max = v
			val = i
		} else if v == max {
			if v > max {
				max = v
				val = i
			}
		}
	}
	fmt.Println(val)
}

File: DataStructure_Alogorithm/Algorithm/Inflearn_basic_cpp/13_가장많이사용된자릿수/main.cpp
#include<stdio.h>
using namespace std;

int ch[10];
int main() {
	int i, digit, max = -214700000, res;
	char a[101];
	scanf("%s", a);
	for (i=0; a[i] != '\0'; i++) {
		digit=a[i] - 48;
		ch[digit]++;
	}

	for(i=0; i<=9; i++) {
		if (ch[i] >= max) {
			max = ch[i];
			res = i;
		}
	}

	printf("%d\n",res);
	return 0;
}